src/Microsoft.AspNet.Mvc.Core/ModelBinding/Validation/DefaultModelValidatorProvider.cs

```diff
@@ -21,7 +21,16 @@ public void GetValidators(ModelValidatorProviderContext context)
                 var validator = context.ValidatorMetadata[i] as IModelValidator;
                 if (validator != null)
                 {
-                    context.Validators.Add(validator);
+                    // Inserts validators based on whether or not they are 'required'. We want to run
+                    // 'required' validators first so that we get the best possible error message.
+                    if (validator.IsRequired)
+                    {
+                        context.Validators.Insert(0, validator);
```

Comment:
I don't think it's important.

> Don't sort validators by IModelValidator.IsRequired. **We should remove this property and have the validator provider put validators in the correct order.** This can avoid duplicating the list and doing the sort using Linq.

Let's just remove `IsRequired`